### PR TITLE
add 'delete task' functionality

### PR DESCRIPTION
**Pull request by VolhaVaitulevich**

### CODE REVIEW

File: src/app/application.ts [hunk 1]
```diff
@@ -111,4 +120,14 @@ export class Application {
       this.renderTasks(this.existingTasks);
     }
   }
+
+  deleteTask(task: ITask): void {
+    const deletedTaskIndex = this.existingTasks.findIndex((item) => item.id === task.id);
```
Comment (siarhei-papkou) [inline src/app/application.ts:125]: filter

File: src/app/application.ts [hunk 2]
```diff
@@ -92,12 +104,9 @@ export class Application {
...
   }
 
-  completeTask(eTarget: ETarget): void {
-    const target = eTarget as HTMLElement;
-    const completedTaskId = Number(target.getAttribute('data_id'));
-
+  completeTask(task: ITask): void {
     //find completed task index to update it
-    const completedTaskIndex = this.existingTasks.findIndex((item) => item.id === completedTaskId);
+    const completedTaskIndex = this.existingTasks.findIndex((item) => item.id === task.id);
```
Comment (siarhei-papkou) [inline src/app/application.ts:109]: map

**Review by siarhei-papkou — CHANGES_REQUESTED** [review]

File: src/app/application.ts [hunk 3]
```diff
@@ -102,43 +109,30 @@ export class Application {
...
+    const newTasksList: Array<ITask> = this.existingTasks.map((item) => {
+      if (item.id === task.id) {
+        item.completed = true;
+      }
+      return item;
+    });
+    newTasksList.sort((item1, item2) => (item1.completed === item2.completed ? 0 : item1.completed ? 1 : -1));
+
+    //add tasks to local storage
+    this.storage.setTasks(newTasksList);
```
Comment (siarhei-papkou) [inline src/app/application.ts:127]: move to updateTasksList

File: src/app/application.ts [hunk 4]
```diff
@@ -102,43 +109,30 @@ export class Application {
...
-    const deletedTaskIndex = this.existingTasks.findIndex((item) => item.id === task.id);
-    if (deletedTaskIndex !== -1) {
-      const newTasksList: Array<ITask> = JSON.parse(JSON.stringify(this.existingTasks));
-      newTasksList.splice(deletedTaskIndex, 1);
-
-      this.storage.setTasks(newTasksList);
-      this.renderTasks(newTasksList);
-    }
+    const newTasksList: Array<ITask> = this.existingTasks.filter((item) => item.id !== task.id);
+    this.storage.setTasks(newTasksList);
```
Comment (siarhei-papkou) [inline src/app/application.ts:135]: move to updateTasksList

File: src/utils/isEqual.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { ITask } from '../app/types';
+
+export function isEqual(existingTasksList: Array<ITask>, newTasksList: Array<ITask>): boolean {
+  console.log(newTasksList);
```
Comment (siarhei-papkou) [inline src/utils/isEqual.ts:4]: remove

File: src/utils/isEqual.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+  if (JSON.stringify(existingTasksList) === JSON.stringify(newTasksList)) {
```
Comment (siarhei-papkou) [inline src/utils/isEqual.ts:5]: image

**Review by siarhei-papkou — CHANGES_REQUESTED** [review]

File: src/utils/isEqual.ts (hunk 6 above)
Comment (VolhaVaitulevich) [reply src/utils/isEqual.ts]: Updated in a new commit

File: src/app/application.ts [hunk 7]
```diff
@@ -8,8 +8,8 @@ export class Application {
 
   constructor(private storage: MyStorage) {
     this.storage = storage;
-    this.existingTasks = storage.getTasks();
-    this.updateTasksList(this.existingTasks);
+    this.existingTasks = [];
```
Comment (siarhei-papkou) [inline src/app/application.ts:11]: image

**Review by siarhei-papkou — APPROVED** [review]

**Pull request merged**